lib/DocblockParser/Parser.php

```diff
@@ -200,10 +201,23 @@ private function parseTypes(): ?TypeNode
             return $type;
         }
         $elements = [$type];
+        $mode = null;
 
         while (true) {
-            if ($this->tokens->if(Token::T_BAR)) {
-                $elements[] = $this->tokens->chomp();
+            if (
+                $this->tokens->if(Token::T_BAR) ||
+                $this->tokens->if(Token::T_AMPERSAND)
+            ) {
+                $delimiter = $this->tokens->chomp();
+                if (!$mode) {
+                    $mode = $delimiter->type;
```

Comment:
so basically a `Foo&Bar|Boo` is invalid and `Boo` will not be parsed. No idea if this is correct or not.

Reply:
i think this should store all the delimiters however so could be "fixed" downstrem